learningequality/studio
# [DO NOT MERGE UNTIL NEW KDS RELEASE] Almost complete icons migration from Vuetify to KDS

## Summary
### Description of the change(s) you made

Almost completes icons migration from Vuetify to KDS.

- Replaces almost all remaining `VIconWrapper`s with `KIcon`, `KLabeledIcon`, or `KIconButton`
- Contains related icon maps and utility function changes
- Replaces `VTooltip` by `KTooltip` at places where KDS icons usage requires it
- Removes global registrations of Vuetify icon 
- Contains few minor UI fixes
- Removes unused code
- Installs https://github.com/learningequality/kolibri-design-system/pull/722 (to be replaced by the KDS release)
  - To fix RTL flip for the `sidebar` icon
  - To be able to teleport `KTooltip` to the teleport root element in few places where there seemed to be no other way to fix its parts being cut off (mostly related to being surrounded by Vuetify components)

See commit messages for details.

Note that the `Thumbnail` component still uses `VIconWrapper`. There are some features that we need to add first to `KIcon` to allow for the refactor. I will try to write what needs to happen to a KDS issue as soon as I have time for this agian. To prevent from using `VIconWrapper` except this one place, I moved it to the `Thumbnail` directory and left deprecation note.
 
### Manual verification steps performed

I was gradually previewing all places where I worked.
___
## Reviewer guidance
### How can a reviewer test these changes?

- Remove `node_modules` (otherwise you won't get updated KDS version in the installation step) and then run `yarn install` to ensure that https://github.com/learningequality/kolibri-design-system/pull/722 work is available
- Do code changes make sense?
- Can you observe any icon related regressions? Note that you don't need to preview all places, as agreed with @bjester, there will be thorough pre-release QA.

## References

Follow-up to @BabyElias's https://github.com/learningequality/studio/pull/4502

----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
